### DISCUSSION TITLE
Ubisys D1 Dimmer switch: configuring the output endpoint

### DISCUSSION THREAD

**Discussion by wb441** [Koenkk/zigbee2mqtt#5343] [Devices]
This is not an issue, rather a lack of knowledge. Can anyone fill me in?
It is about the [Ubisys D1](https://www.zigbee2mqtt.io/devices/D1.html) which communicates to a RPi via the CC2531.
![image](https://user-images.githubusercontent.com/40491548/102651536-feadab80-416c-11eb-8bfe-4f5848bd37ad.png)

Problem: 
When switching the connected light on or off, the device takes a certain time to dim up/down before being fully on or off.
This takes too long and doesn't feel 'snappy'

The manufacturer of the Ubisys D1 has a nice file documenting the device. [Link](https://www.ubisys.de/wp-content/uploads/ubisys-d1-technical-reference.pdf)
I want to set the parameter listed on page 18 of this document under **7.2.6. Level Control Cluster (Server)** , more specifically OnOffTransitionTime.

I've been reading the instructions on this device under Supported devices, however, these go in full detail how to configure the **input** endpoints (pretty cool!). But not how to configure the **output** endpoints. The setting I want to change belongs to the output endpoint that is the dimmed load.

How can I set OnOffTransitionTime to 0 using Zigbee2mqtt? 
I've been trying to analyze the files from Zigbee-herdsman  (devices.js, toZigbee, fromZigbee) and am sorta stuck.

Thanks in advance for any pointers!

**Comment by sjorge** [comment]
https://www.zigbee2mqtt.io/devices/C4.html

Has some docs that applies to S1 and S2, I'd assume also to the D1...
Configuration of the devices happens via a special ep 232, so which is used for device provisioning.

If one of the templates mentioned in the docs doesn't work for you, you might have to play with the raw input configuration. (Which is kind of a pain).
